crates/ruff_python_formatter/src/statement/suite.rs

```diff
@@ -59,287 +59,269 @@ impl FormatRule<Suite, PyFormatContext<'_>> for FormatSuite {
         let source = f.context().source();
         let source_type = f.options().source_type();
 
-        let mut f = WithNodeLevel::new(node_level, f);
-        write!(
-            f,
-            [format_with(|f| {
-                let mut iter = statements.iter();
-                let Some(first) = iter.next() else {
-                    return Ok(());
-                };
-
-                // Format the first statement in the body, which often has special formatting rules.
-                let first = match self.kind {
-                    SuiteKind::Other => {
-                        if is_class_or_function_definition(first) && !comments.has_leading(first) {
-                            // Add an empty line for any nested functions or classes defined within
-                            // non-function or class compound statements, e.g., this is stable formatting:
-                            // ```python
-                            // if True:
-                            //
-                            //     def test():
-                            //         ...
-                            // ```
-                            empty_line().fmt(f)?;
-                        }
-
-                        SuiteChildStatement::Other(first)
-                    }
+        let mut with_node_level = WithNodeLevel::new(node_level, f);
+        let f = &mut with_node_level;
```

Comment:
Is the `&mut` assignment necessary? Most call sites simply use `let mut f = WithNodeLevel::new(node_level, f)`

Reply:
To my surprise `let f = &mut WithNodeLevel::new(node_level, f);` works, i'd have expected that it wants a let binding for that